static/app/components/globalDrawer/context.tsx

```diff
@@ -0,0 +1,69 @@
+import {createContext, useCallback, useContext, useState} from 'react';
+
+import type {DrawerOptions, DrawerRenderProps} from 'sentry/components/globalDrawer';
+import GlobalDrawer from 'sentry/components/globalDrawer';
+
+type DrawerRenderer = (renderProps: DrawerRenderProps) => React.ReactNode;
+
+export interface DrawerConfig {
+  renderer: DrawerRenderer | null;
+  options?: DrawerOptions;
+}
+
+interface DrawerContextProps {
+  closeDrawer: () => void;
+  config: DrawerConfig;
+  openDrawer: (
+    renderer: DrawerConfig['renderer'],
+    options?: DrawerConfig['options']
+  ) => void;
+}
+
+const DEFAULT_DRAWER_CONTEXT: DrawerContextProps = {
+  config: {
+    renderer: null,
+    options: {
+      closeOnEscapeKeypress: true,
+      closeOnOutsideClick: true,
+    },
+  },
+  openDrawer: () => {},
+  closeDrawer: () => {},
+};
+
+export const DrawerContext = createContext<DrawerContextProps>(DEFAULT_DRAWER_CONTEXT);
+
+interface DrawerContextProviderProps {
+  children: React.ReactNode;
+}
+
+export function DrawerContextProvider({children}: DrawerContextProviderProps) {
+  const [drawerConfig, setDrawerConfig] = useState<DrawerConfig>(
+    DEFAULT_DRAWER_CONTEXT.config
+  );
+  const openDrawer = useCallback(
+    (renderer, options = {}) => setDrawerConfig({renderer, options}),
+    [setDrawerConfig]
+  );
+  const closeDrawer = useCallback(
+    () => setDrawerConfig(DEFAULT_DRAWER_CONTEXT.config),
+    [setDrawerConfig]
+  );
+
+  const ctx: DrawerContextProps = {
+    config: drawerConfig,
+    closeDrawer,
+    openDrawer,
+  };
+
+  return (
+    <DrawerContext.Provider value={ctx}>
+      <GlobalDrawer />
```

Comment:
is there any way to not render the global drawer until its used for the first time. mostly thinking about the rtl tests

Reply:
Yeah a bunch of tests were failing, I added an option (`includeDrawer`) to explcitly render the drawer when using `render`. I'm not sure how obvious it will be to find as someone writing tests, but it fixes all the tests failing from `toBeEmptyDOMElement` assertions.

I saw we were already doing something like this with organization/router context, but if this feels hacky, lmk.